documentation/APIs/core-models.md

```diff
@@ -0,0 +1,588 @@
+            <td>The Block ID of the parent.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td>Issuer ID</td>
+    <td>ByteArray[32]</td>
+    <td>The issuer identifier</td>
+  </tr>
+  <tr>
+    <td>Issuing Time</td>
+    <td>uint64</td>
+    <td>The time the block is issued. It's a Unix-like timestamp in nanosecond.</td>
+  </tr>
+  <tr>
+    <td valign="top">Slot Commitment <code>oneOf</code></td>
+    <td colspan="2">
+      <details open="true">
+        <summary>Slot Commitment</summary>
+        <blockquote>
+          Slot Commitment is an object that contains a summary of a slot. More descriptions are in Slot Commitment section.
+        </blockquote>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td>Latest Finalized Slot</td>
+    <td>uint64</td>
+    <td>The slot index of latest finalized slot.</td>
+  </tr>
+  <tr>
+    <td>Payload Length</td>
+    <td>uint32</td>
+    <td>The length of the following payload in bytes. A length of 0 means no payload will be attached.</td>
+  </tr>
+  <tr>
+    <td valign="top">Payload <code>optOneOf</code></td>
+    <td colspan="2">
+      <details>
+        <summary>Tagged Data Payload</summary>
+        <blockquote>
+          With Payload Type 5, more details are described in Tagged Data Payload section.
+        </blockquote>
+      </details>
+      <details>
+        <summary>Transaction Payload</summary>
+        <blockquote>
+          With Payload Type 6, more details are described in Transaction section.
+        </blockquote>
+      </details>
+  </tr>
+  <tr>
+    <td valign="top">Signature <code>oneOf</code></td>
+    <td colspan="2">
+      <details>
+        <summary>Ed25519 Signature</summary>
+        <blockquote>
+          With Signature Type 0, more details are described in Signature section.
+        </blockquote>
+      </details>
+    </td>
+    </td>
+  </tr>
+  <tr>
+    <td>Nonce</td>
+    <td>uint64</td>
+    <td>The nonce which lets this block fulfill the PoW requirement.</td>
+  </tr>
+</table>
+
+### Block Signature
+This is how the signature is generated:
+
+* `content` is the serialized block **without** signature and nonce.
+* `slot_commitment_ID` is the identifier calculated from the Slot Commitment within the block, following steps described in [Slot Commitment ID](#slot-commitment-id) section.
+* `Issuing Time` is the block issuing time.
+
+Calculation:
+1. `content_hash` = hash(`content`)
+2. `data` = Concat(`Issuing Time`, `slot_commitment_ID`, `content_hash`)
+3. And sign with the Issuer PubKey, `Signature` = Sign(`data`) 
+
+### Block ID
+Block ID denotes an identifier of a block, with type `ArrayBytes[40]`. It is calculated as the following steps, using BLAKE2b-256 hash function:
+
+* `content` is the serialized block **without** signature and nonce.
+* `slot_index` is the slot index of the `Issuing Time` of the block.
+   **Note**: It's **not** the same slot index as in the Slot Commitment.
```

Comment:
So it means that we can't deduce the BlockID just by looking at the Block?

Reply:
We're having a discussion on this but might have an answer by next Wednesday. :cry: